nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -1056,22 +637,61 @@ ngx_http_modsecurity_thread_completion(ngx_event_t *ev)
 }
 
 
+static void
+ngx_http_modsecurity_detection_thread_func(void *data, ngx_log_t *log)
+{
+    // Executed in a separate thread
+    ngx_http_modsecurity_detection_thread_ctx_t *thread_ctx = data;
+
+    // Processing request headers
+    ngx_int_t rc = ngx_http_modsecurity_status(modsecProcessRequestHeaders(thread_ctx->req));
+    if (rc != NGX_DECLINED) {
+        return;
+    }
+
+    if (modsecContextState(thread_ctx->req) == MODSEC_DISABLED) {
+        return;
+    }
+
+    // The name of modsecProcessRequestBody is a bit misleading. This function call is needed even to just process GET args.
```

Comment:
I am following the original logic where `modsecProcessRequestBody` is called unconditionally.
The conditional part is where we call `ngx_http_modsecurity_load_request_body` only if `modsecIsRequestBodyAccessEnabled` is `true`, but `modsecProcessRequestBody` has always been called regardless of body access being enabled.

Do you think we should alter the logic in detection branch?

Reply:
ok, you have moved logic out, that's why i am confused at the first place. that should be fine

Reply:
Yeah, some logic duplication takes place but I found it easier to keep that way rather than try to unite them artificially. But the concept behind the logic remains the same.